sdk/storage/Azure.Storage.Files/src/FileUriBuilder.cs

```diff
@@ -119,6 +120,13 @@ public SasQueryParameters Sas
         }
         private SasQueryParameters _sas;
 
+        /// <summary>
+        /// Get the terminal piece from the <see cref="DirectoryOrFilePath"/>, or null if
+        /// not present in the <see cref="Uri"/>.
+        /// </summary>
+        internal string TerminalPiece =>
```

Comment:
I don't love this name.  What's a piece?  Can we call it something more like `CurrentDirectoryOrFileName`?

Reply:
How about TerminalDirectoryOrFileName? I think using Current is somewhat ambiguous.

Reply:
I'd rather use `Last` than `Terminal` because that's more consistent with .NET naming and the directory isn't dying.  I'd also slightly prefer `GetLastDirectoryOrFileName()` as a method since we're doing something nontrivial, but don't care as much since it's internal.

Reply:
I think Terminal is more precise than Last since Last can mean something like previous, and this is an internal property so I figured precision was more important than accessibility. That said, I don't really feel too strongly about this so I will go ahead and change it.